pkg/controller/unreachable/unreachable_controller.go

```diff
@@ -87,7 +87,7 @@ func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) rec
 func AddToManager(mgr manager.Manager, r reconcile.Reconciler, concurrentReconciles int, rateLimiter workqueue.RateLimiter) error {
 	// Create a new controller
 	c, err := controller.New("unreachable-controller", mgr, controller.Options{
-		Reconciler:              r,
+		Reconciler:              controllerutils.NewDelayingReconciler(r, log.WithField("controller", ControllerName)),
```

Comment:
Notwithstanding the naïf example you followed from my early days with k8s, I think this wrapping should actually be done in the NewReconciler func above. Similar for other controllers.

Reply:
The reason I don't wrap in the `NewReconciler` func is that some later functions expect the unwrapped version of the reconciler, such as the `clusterdeployment` controller (which casts the reconciler interface back into the concrete `ReconcileClusterDeployment` type)

Reply:
:shrug: okay